boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -286,6 +292,23 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx < ENTRY_MIN_SIZE || Regs.x.ecx > sizeof(BIOS_MEMORY_MAP))
+        {
+            ERR("Int 15h AX=E820h returned an invalid entry length! (would-be-PcBiosMapCount = %lu, Entry length = (%Iu <=) %lu (<= %Iu))\n\n",
+                PcBiosMapCount, ENTRY_MIN_SIZE, Regs.x.ecx, sizeof(BIOS_MEMORY_MAP));
+            /* Warn user, unless wrong case is "first and too small entry", which is otherwise harmless. */
+            if (PcBiosMapCount > 0 || Regs.x.ecx > sizeof(BIOS_MEMORY_MAP))
+                ASSERTMSG("Int 15h AX=E820h returned an invalid entry length!", FALSE);
```

Comment:
Oh, forgot about this the first time around: braces around all if blocks please.

Reply:
Done.